hw/ip/otbn/data/insns.yml

```diff
@@ -947,18 +955,18 @@ insns:
           The direction of an optional shift applied to `<wrs2>`.
       - &bn-shift-bytes-operand
         name: shift_bytes
-        type: imm5
+        type: uimm5
```

Comment:
Won't we need this change for `bn.addc`, `bn.sub` and `bn.subb` plus the RV32I shift instructions?

Reply:
I don't think I understand. `bn.addc` has the same operands as `bn.add` (because it uses `operands: *bn-add-operands` in the file). `bn.sub` grabs the shift type from `bn.add` and then `bn.subb` has the same operands as `bn.sub`.

The RV32I shift instructions with immediates use `*shamt-operand`, defined by `slli`, which declares the type as `uimm`.

I intentionally changed from `imm` to `simm` or `uimm` to make sure the tooling would explode if I missed anything :-)

Reply:
Ah right hadn't realised this would replicate across to `bn.addc` `bn.sub` and `bn.subb` and missed the `shamt-operand` change.